### PR TITLE
Add type definition for remark-heading-id

### PR DESCRIPTION
**Pull request by drpsyko101**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/remark-heading-id/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "rules": {
+        "@definitelytyped/no-outside-dependencies": "off"
```
Comment (jakebailey) [inline types/remark-heading-id/.eslintrc.json:3]: This should not be turned off.

**Review by jakebailey — CHANGES_REQUESTED** [review]

**Review by imcuttle — APPROVED** [review]
Approved! Thanks

File: types/remark-heading-id/.eslintrc.json (hunk 1 above)
Comment (drpsyko101) [reply types/remark-heading-id/.eslintrc.json]: Removed the `.eslintrc` rule in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68755/commits/7bb4f6919d43df8052cd71c2012bb249805cb9ec

**Review by jakebailey — APPROVED** [review]

**Review by imcuttle — APPROVED** [review]

**Comment by drpsyko101** [comment]
Ready to merge. Thanks @imcuttle and @jakebailey for the review.

**Pull request merged**